web/ui/react-app/src/QueryStats.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { PureComponent } from 'react';
+import './QueryStats.css';
+
+interface QueryStatsProps {
+    stats: {
+        loadTime: number;
+        resolution: number;
+        totalTimeSeries: number;
+    } | null;
+    loading: boolean;
+}
+
+class QueryStats extends PureComponent<QueryStatsProps>{
+
+    render() {
+        const stats = this.props.stats;
+        const loading = this.props.loading;
+
+        return (
+            <div className="query-stats">
```

Comment:
@CSTDev is it needed this component to be created at all if !loading && stats !== null?

Reply:
Fair point will do the check earlier.